dookug-document/dookug-document-service/src/main/java/hu/icellmobilsoft/dookug/document/service/action/DocumentSignAction.java

```diff
@@ -0,0 +1,109 @@
+import jakarta.inject.Inject;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.StreamingOutput;
+
+import com.github.jknack.handlebars.internal.lang3.StringUtils;
+
+import hu.icellmobilsoft.coffee.dto.exception.InvalidParameterException;
+import hu.icellmobilsoft.coffee.dto.exception.enums.CoffeeFaultType;
+import hu.icellmobilsoft.coffee.se.api.exception.BaseException;
+import hu.icellmobilsoft.coffee.se.api.exception.TechnicalException;
+import hu.icellmobilsoft.dookug.api.dto.exception.IOExceptionBaseExceptionWrapper;
+import hu.icellmobilsoft.dookug.api.rest.document.form.DocumentSignMultipartForm;
+import hu.icellmobilsoft.dookug.engine.pdfbox.signing.SignatureGenerator;
+import hu.icellmobilsoft.dookug.schemas.document._1_0.rest.documentsign.DocumentSignRequest;
+
+/**
+ * PDF electronic signing action
+ *
+ * @author tamas.cserhati
+ * @since 1.1.0
+ */
+@Model
+public class DocumentSignAction extends BaseDocumentGenerateAction {
+
+    private static final String ATTACHMENT_FILENAME = "attachment; filename=\"{0}\"";
+    private static final String DEFAULT_PDF_NAME = "signed-document.pdf";
+
+    @Inject
+    private SignatureGenerator signatureGenerator;
+
+    /**
+     * Multipart form based document signing. The form contains all required data.
+     *
+     * @param form
+     *            {@link DocumentSignMultipartForm} The multipart form object
+     * @return Signed document
+     * @throws BaseException
+     *             if any error occurs
+     */
+    public Response postSignDocumentMultipart(DocumentSignMultipartForm form) throws BaseException {
+        if (form == null) {
+            throw new InvalidParameterException("form is null!");
+        }
+        if (form.getRequest() == null || !form.getRequest().isSetDigitalSignatureProfile()) {
+            throw new InvalidParameterException("Request part is invalid!");
+        }
+        try {
+            OutputStream outputStream = signatureGenerator.getOutputStreamForUnsignedPdf();
+            // saving input stream to a temporaly file
+            form.getDocument().transferTo(outputStream);
+            // create signed pdf response
+            StreamingOutput streamingOutput = createDocumentStreamingOutput(form.getRequest());
+            return Response.ok(streamingOutput)
+                    .header("Content-Disposition", getContentDispositionHeaderValue(form.getRequest().getFileName()))
+                    .build();
+        } catch (IOException e) {
+            throw new TechnicalException(
+                    CoffeeFaultType.OPERATION_FAILED,
+                    MessageFormat.format("Error occured while working with input stream: [{0}]", e.getLocalizedMessage()));
+        }
+    }
+
+    private String getContentDispositionHeaderValue(String fileName) {
+        return MessageFormat.format(ATTACHMENT_FILENAME, StringUtils.isBlank(fileName) ? DEFAULT_PDF_NAME : fileName);
```

Comment:
ha küldenek fileName-et:
- .pdf kiterjesztést vagy adjuk hozzá ,
- vagy kérjük be a fileName-el
xsd description-ben, és végpont leírásnál egyértelmüsítsük